<?xml version="1.0" encoding="UTF-8"?>
<!--This does not replace URDF, and is not an extension of URDF.
    This is a format for representing semantic information about the robot structure.
    A URDF file must exist for this robot as well, where the joints and the links that are referenced are defined
-->
<robot name="hc10_with_hand">
    <!--GROUPS: Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
    <!--LINKS: When a link is specified, the parent joint of that link (if it exists) is automatically included-->
    <!--JOINTS: When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
    <!--CHAINS: When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
    <!--SUBGROUPS: Groups can also be formed by referencing to already defined group names-->
    <group name="arm">
        <joint name="joint_1_s"/>
        <joint name="joint_2_l"/>
        <joint name="joint_3_u"/>
        <joint name="joint_4_r"/>
        <joint name="joint_5_b"/>
        <joint name="joint_6_t"/>
    </group>
    <group name="hand">
        <joint name="arm_to_hand_frame"/>
        <joint name="fing1"/>
        <joint name="fing2"/>
    </group>
    <!--GROUP STATES: Purpose: Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
    <group_state name="look_camera" group="arm">
        <joint name="joint_1_s" value="0.5019"/>
        <joint name="joint_2_l" value="-0.0558"/>
        <joint name="joint_3_u" value="-0.0872"/>
        <joint name="joint_4_r" value="0.2788"/>
        <joint name="joint_5_b" value="1.8031"/>
        <joint name="joint_6_t" value="0"/>
    </group_state>
    <group_state name="open" group="hand">
        <joint name="fing1" value="0.05"/>
        <joint name="fing2" value="0.05"/>
    </group_state>
    <group_state name="close" group="hand">
        <joint name="fing1" value="0"/>
        <joint name="fing2" value="0"/>
    </group_state>
    <group_state name="home" group="arm">
        <joint name="joint_1_s" value="0"/>
        <joint name="joint_2_l" value="0"/>
        <joint name="joint_3_u" value="0"/>
        <joint name="joint_4_r" value="0"/>
        <joint name="joint_5_b" value="0"/>
        <joint name="joint_6_t" value="0"/>
    </group_state>
    <!--END EFFECTOR: Purpose: Represent information about an end effector.-->
    <end_effector name="hand" parent_link="link_6_t" group="hand" parent_group="arm"/>
    <!--DISABLE COLLISIONS: By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->
    <disable_collisions link1="base_link" link2="link_1_s" reason="Adjacent"/>
    <disable_collisions link1="base_link" link2="link_3_u" reason="Never"/>
    <disable_collisions link1="base_link" link2="part_1" reason="Adjacent"/>
    <disable_collisions link1="camera_sup2" link2="ffld_0850ncn30_lksr00_0__3" reason="Never"/>
    <disable_collisions link1="camera_sup2" link2="ffld_0850ncn30_lksr00_0__3_2" reason="Never"/>
    <disable_collisions link1="camera_sup2" link2="ffld_0850ncn30_lksr00_0__6" reason="Adjacent"/>
    <disable_collisions link1="camera_sup2" link2="link_3_u" reason="Never"/>
    <disable_collisions link1="camera_sup2" link2="link_5_b" reason="Never"/>
    <disable_collisions link1="camera_sup2" link2="link_6_t" reason="Never"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__3" link2="ffld_0850ncn30_lksr00_0__3_2" reason="Default"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__3" link2="ffld_0850ncn30_lksr00_0__6" reason="Adjacent"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__3" link2="link_3_u" reason="Never"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__3" link2="link_4_r" reason="Never"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__3" link2="link_5_b" reason="Never"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__3" link2="link_6_t" reason="Never"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__3_2" link2="ffld_0850ncn30_lksr00_0__6" reason="Adjacent"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__3_2" link2="link_3_u" reason="Never"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__3_2" link2="link_4_r" reason="Never"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__3_2" link2="link_5_b" reason="Never"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__3_2" link2="link_6_t" reason="Never"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__6" link2="link_3_u" reason="Never"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__6" link2="link_4_r" reason="Never"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__6" link2="link_5_b" reason="Never"/>
    <disable_collisions link1="ffld_0850ncn30_lksr00_0__6" link2="link_6_t" reason="Adjacent"/>
    <disable_collisions link1="link_1_s" link2="link_2_l" reason="Adjacent"/>
    <disable_collisions link1="link_1_s" link2="link_3_u" reason="Never"/>
    <disable_collisions link1="link_1_s" link2="part_1" reason="Never"/>
    <disable_collisions link1="link_2_l" link2="link_3_u" reason="Adjacent"/>
    <disable_collisions link1="link_2_l" link2="link_4_r" reason="Never"/>
    <disable_collisions link1="link_3_u" link2="link_4_r" reason="Adjacent"/>
    <disable_collisions link1="link_3_u" link2="link_5_b" reason="Never"/>
    <disable_collisions link1="link_3_u" link2="link_6_t" reason="Never"/>
    <disable_collisions link1="link_4_r" link2="link_5_b" reason="Adjacent"/>
    <disable_collisions link1="link_4_r" link2="link_6_t" reason="Never"/>
    <disable_collisions link1="link_5_b" link2="link_6_t" reason="Adjacent"/>
</robot>
